File: memory-domain/src/main/java/com/memory/domain/calendar/repository/AnniversaryEventRepositoryCustom.java
package com.memory.domain.calendar.repository;

import com.memory.domain.calendar.AnniversaryEvent;
import com.memory.domain.member.Member;

import java.time.LocalDateTime;
import java.util.List;

public interface AnniversaryEventRepositoryCustom {

    List<AnniversaryEvent> findByMemberAndStartDateTimeBetween(Member member, LocalDateTime startDate, LocalDateTime endDate);

}

File: memory-domain/src/main/kotlin/com/memory/domain/calendar/repository/AnniversaryEventRepositoryCustom.kt
package com.memory.domain.calendar.repository

import com.memory.domain.calendar.AnniversaryEvent
import com.memory.domain.member.Member
import java.time.LocalDateTime

interface AnniversaryEventRepositoryCustom {

    fun findByMemberAndStartDateTimeBetween(member: Member?, startDate: LocalDateTime?, endDate: LocalDateTime?): List<AnniversaryEvent>

    fun findByMemberAndIsDdayTrue(member: Member?): List<AnniversaryEvent>

}
